### PR TITLE
Clean up libtestdriver1 config: unconditonal removals

### LINKED ISSUE #8488: Clean up libtestdriver1 config: unconditonal removals
The file `crypto_config_test_driver_extension.h` is use to reverse `ACCEL` macros when building `libtestdriver1`: when something is accelerated in the main library, it should be built-in in libtestdriver1, and when something's not accelerated in the main library, we don't need it in libtestdriver1, so we set the corresponding `ACCEL` macro, causing the built-in implementation to be omitted from libtestdriver1, in order to keep it minimal (we can't use `PSA_WANT` symbols for that, as they need to be the same between `libtestdriver1` and the main libraries, in order for them to use the same ABI).

So, the file consists mostly of blocks that check the state of an `ACCEL` macro and reverses it.

However [near the end, some `ACCEL` macros are set unconditionally](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/include/test/drivers/crypto_config_test_driver_extension.h#L524). There is not comment explaining why, but it seems the intention was to exclude things that can't be accelerated yet anyway. However:
- the list does not seem up-to-date: it has GCM and CCM for example which can clearly be accelerated (we has tests for that);
- it doesn't seem to always work as intended as we can build `libtestdriver1` with GCM and CCM included (again, we had tests relying on that).

This task is to clean this up. For each macro in this final block, either:
- quit defining `ACCEL` unconditionally and just keep the normal reversing logic for this macro;
- or keep defining `ACCEL` unconditionally but add a comment explaining why.

While at it, maybe reword the initial comment in this file: I think "mirror" is ambiguous (does it mean "the same" or "the opposite" and it wouldn't hurt to explain why we're doing this (the 1st paragraph of this issue's description could be used as a starting point).

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

Resolves #8488.

## PR checklist

- [x] **changelog** not required - test utilities only
- [x] **backport** not required - only matters for new development
- [x] **tests** same as the changelog: no symbol or code change was required to support this improvement, so there is no need for new tests

### CODE REVIEW

**Comment by valeriosetti** [comment]
I removed the `need-ci` label because the only failing test in OpenCI was `Windows-2013` but it seems to me that this is more related to the CI rather than the code. In support of this InternCI fully passed.

**Review by mpg — APPROVED** [review]
LGTM

**Review by lpy4105 — APPROVED** [review]
LGTM
